src/animation/tracks/QuaternionKeyframeTrack.js

```diff
@@ -5,29 +5,24 @@ import { QuaternionLinearInterpolant } from '../../math/interpolants/QuaternionL
 /**
  * A Track of quaternion keyframe values.
  */
+class QuaternionKeyframeTrack extends KeyframeTrack {
 
-function QuaternionKeyframeTrack( name, times, values, interpolation ) {
+	InterpolantFactoryMethodLinear( result ) {
 
-	KeyframeTrack.call( this, name, times, values, interpolation );
+		return new QuaternionLinearInterpolant( this.times, this.values, this.getValueSize(), result );
 
-}
+	}
 
-QuaternionKeyframeTrack.prototype = Object.assign( Object.create( KeyframeTrack.prototype ), {
+}
 
-	constructor: QuaternionKeyframeTrack,
+Object.assign( QuaternionKeyframeTrack.prototype, {
 
 	ValueTypeName: 'quaternion',
 
 	// ValueBufferType is inherited
 
 	DefaultInterpolation: InterpolateLinear,
 
-	InterpolantFactoryMethodLinear: function ( result ) {
-
-		return new QuaternionLinearInterpolant( this.times, this.values, this.getValueSize(), result );
-
-	},
-
 	InterpolantFactoryMethodSmooth: undefined // not yet implemented
```

Comment:
I noticed this pattern where prototype methods are undefined to drive fallback logic in base `KeyframeTrack`. It feels kind of wrong to mutate the KeyframeTrack interface, although I guess if these can be considered internal it doesn't matter. No need to change now, just pointing it out as another candidate for some cleanup down the road.